src/pages/LoginPage.tsx

```diff
@@ -1,55 +1,66 @@
-import React, { Component, FormEvent, ChangeEvent } from 'react';
-import SessionInfo from '../types';
-import { Redirect } from 'react-router';
+import React, { Component, FormEvent, ChangeEvent } from "react";
+import SessionInfo from "../types";
+import { Redirect } from "react-router";
 
 class LoginPage extends Component {
-    readonly state: {
-        email: string | null,
-        password: string | null,
-        sessionInfo: SessionInfo | null,
-    } = {
-        email: null,
-        password: null,
-        sessionInfo: null,
-    };
+  readonly state: {
+    email: string | null;
```

Comment:
in ts are you supposed to terminate w/ `;`s in a type definition? Don't think I've seen a mix of the syntaxes before

Reply:
ya, type and interface definitions are terminated with semicolons - it's like doing
```
interface LoginState {
  email: string | null;
  password: string | null;
}

class LoginPage extends Component {
  readonly state: LoginState = { ... }
  ...
```